commands/dataset/list/mod.ts

```diff
@@ -0,0 +1,70 @@
+import { datasets } from "../../../api/dataset.ts";
+import { config } from "../../../config.ts";
+import * as psFlags from "../../../flags.ts";
+import { asserts } from "../../../lib/asserts.ts";
+import { dataTable } from "../../../lib/data-table.ts";
+import { loading } from "../../../lib/loading.ts";
+import { pickJson } from "../../../lib/pick-json.ts";
+import { command, flag, flags } from "../../../zcli.ts";
+import { defaultFields } from "../mod.ts";
+
+/**
+ * This variable is automatically generated by `zcli add`. Do not remove this
+ * or change its name unless you're no longer using `zcli add`.
+ */
+const subCommands: ReturnType<typeof command>[] = [];
+
+export const list = command("list", {
+  short: "List datasets.",
+  long: ({ root }) => `
+    List datasets in your team or project.
+
+    Pick a subset of fields to display:
+    \`\`\`
+    ${root.name} dataset list -F name -F dtModified
+    \`\`\`
+  `,
+  commands: subCommands,
+  flags: psFlags.paginator.merge(flags({
+    "name": flag({
+      aliases: ["n"],
+      short: "Filter by name.",
+    }).ostring(),
+  })),
+  // We use command metadata in the `persistentPreRun` function to check if a
+  // command requires an API key. If it does, we'll check to see if one is
+  // set. If not, we'll throw an error.
+  meta: {
+    requireApiKey: true,
+  },
+}).run(
+  async function* ({ flags }) {
+    const team = await config.get("team");
+    asserts(team, "You must be in a team to list datasets.");
+    const result = await loading(
+      datasets.list({
+        limit: flags.limit,
+        after: flags.after,
+        orderBy: "name",
```

Comment:
Test is faling on this; not sure if the underlying type needs to be updated to support that ordering (and whether it requires back-end changes)

Reply:
(maybe ordering by the supported dtCreated would be the least-suffering path? The order doesn't seem to be documented, so we could arguably change that in the future without it being considered a breaking change, and also arguably the date is as good as the name as a command output)